wlm/src/main/java/ai/djl/serving/wlm/SecureModeUtils.java

```diff
@@ -0,0 +1,389 @@
+        Path servingPropertiesFile = lookForFile(directory, "serving.properties");
+        if (servingPropertiesFile != null) {
+            List<String> lines = Files.readAllLines(servingPropertiesFile);
+            for (String line : lines) {
+                if (line.toLowerCase().startsWith(option.toLowerCase())) {
+                    servingPropertiesValue = line.split("=")[1].trim();
+                }
+            }
+        }
+        // Handle disallowed value on a per-option basis
+        if ("option.trust_remote_code".equals(option)) {
+            // The disallowed value is boolean "true"
+            if (servingPropertiesValue != null && Boolean.parseBoolean(servingPropertiesValue)) {
+                logErrorAndExit(
+                        "Setting option.trust_remote_code to True is prohibited in Secure Mode."
+                                + " Exiting early.");
+            }
+        } else if ("option.entryPoint".equals(option)) {
+            // The disallowed value is a .py file
+            if (servingPropertiesValue != null && servingPropertiesValue.endsWith(".py")) {
+                logErrorAndExit(
+                        "Setting option.entryPoint to custom Python script is prohibited in Secure"
+                                + " Mode. Exiting early.");
+            }
+        } else {
+            throw new IllegalArgumentException("Invalid disallowed option: " + option);
+        }
+    }
+
+    /**
+     * Search for a tokenizer_config.json in the directory. If found, check if Jinja chat_template
+     * key is present. If so, fast-fail.
+     *
+     * @param directory the directory to search for tokenizer_config.json
+     * @throws IOException if there is an error walking the directory
+     */
+    private static void scanForChatTemplate(Path directory) throws IOException {
+        Path tokenizerConfig = lookForFile(directory, "tokenizer_config.json");
+        if (tokenizerConfig != null) {
+            try (FileReader fileReader = new FileReader(tokenizerConfig.toString())) {
+                JsonObject jsonObject = JsonParser.parseReader(fileReader).getAsJsonObject();
+                if (jsonObject.has("chat_template")) {
+                    logErrorAndExit(
+                            "Jinja chat_template field found in "
+                                    + tokenizerConfig.toString()
+                                    + ", but is prohibited in Secure Mode. Exiting early.");
+                }
+            }
+        }
+    }
+
+    /**
+     * Link additional requirements.txt files into requirements.txt in modelDir using -r. This
+     * single requirements.txt will be installed during Python engine initialization.
+     *
+     * @param pathList list of absolute paths
+     * @param modelDir path to model_dir
+     * @throws IOException
+     */
+    private static void linkAdditionalRequirementsTxt(List<String> pathList, Path modelDir)
```

Comment:
what was this function for?

Reply:
In case some artifacts in trusted additional model data paths need special requirements, they will be gathered together and installed in the pyenv. I don't believe there is any specific need currently but possibly in the future. I'm considering leaving it out for now